tests/tx/test_scripts.py

```diff
@@ -972,3 +982,105 @@ def test_get_sigops_count(self):
         self.assertEqual(get_sigops_count(input_script.data, p2pkh_script), 1)
         # if no output_script, count only input
         self.assertEqual(get_sigops_count(input_script.data), 1)
+
+    def test_op_sighash_bitmask(self) -> None:
+        with self.assertRaises(MissingStackItems):
+            op_sighash_bitmask(ScriptContext(stack=[], extras=Mock(), logs=[], settings=Mock()))
+
+        with self.assertRaises(MissingStackItems):
+            op_sighash_bitmask(ScriptContext(stack=[b''], extras=Mock(), logs=[], settings=Mock()))
+
+        with self.assertRaises(AssertionError):
+            op_sighash_bitmask(ScriptContext(stack=[0b111, 0b101], extras=Mock(), logs=[], settings=Mock()))
+
+        stack: list[bytes | int | str] = [bytes([0b0]), bytes([0xFF, 0xFF])]
+        context = Mock(spec_set=ScriptContext)
+        extras = Mock(spec_set=ScriptExtras)
+        context.stack = stack
+        context.extras = extras
+
+        with self.assertRaises(CustomSighashModelInvalid):
+            op_sighash_bitmask(context)
+
+        context.stack = [bytes([0b111]), bytes([0b101])]
```

Comment:
Maybe add tests with different endianness.

Reply:
What do you mean? Endianness has no meaning in this context as this string of bits is never interpreted as a number, it's simply a string of bits

Reply:
I meant that `0b1101` should be different from `0b1011`.